packages/dd-trace/src/appsec/index.js

```diff
@@ -225,12 +222,48 @@ function onPassportVerify ({ credentials, user }) {
   passportTrackEvent(credentials, user, rootSpan, config.appsec.eventTracking.mode)
 }
 
+const responseAnalyzedSet = new WeakSet()
+const responseBlockedSet = new WeakSet()
+
+function onResponseWriteHead ({ req, res, abortController, statusCode, responseHeaders }) {
+  // avoid "write after end" error
+  if (responseBlockedSet.has(res)) {
+    abortController?.abort()
+    return
+  }
+
+  // avoid double waf call
+  if (responseAnalyzedSet.has(res)) {
+    return
+  }
+
+  const rootSpan = web.root(req)
+  if (!rootSpan) return
+
+  responseHeaders = Object.assign({}, responseHeaders)
+  delete responseHeaders['set-cookie']
```

Comment:
why? :)

Reply:
cookies are sent in different address, you could see it in [L130](https://github.com/DataDog/dd-trace-js/pull/3837/files#diff-bfb3e447c35a0c6f6b4131ecef93afce24a08ef37f11fe6587c74be9c86c9920L135)

This code is copied from [old L114](https://github.com/DataDog/dd-trace-js/pull/3837/files#diff-bfb3e447c35a0c6f6b4131ecef93afce24a08ef37f11fe6587c74be9c86c9920L114)

Reply:
The cookies have vastly different attack patterns, so it has its own address, it's been like this forever since sqreen times

Reply:
the name of headers address is `server.respones.headers.no_cookies` so obviously, no cookies